astropy/modeling/core.py

```diff
@@ -3341,6 +3341,79 @@ def render(self, out=None, coords=None):
         return out
 
 
+    def replace_submodel(self, name=None, model=None):
```

Comment:
Is the default `model=None` checked? I wonder if it can't be used for dropping a model (in some future version of the code).

Reply:
I've removed the default `None` for both parameters since neither is meaningful, so no test required. If `replace_submodel()` gets handled via `__setitem()__` (as suggested) then `__delitem()__` should probably deal with dropping a model.